parquet-column/src/main/java/org/apache/parquet/schema/LogicalTypeAnnotation.java

```diff
@@ -1183,6 +1234,136 @@ public boolean equals(Object obj) {
     }
   }
 
+  public static class GeometryLogicalTypeAnnotation extends LogicalTypeAnnotation {
+    private final String crs;
+
+    private GeometryLogicalTypeAnnotation(String crs) {
+      this.crs = crs;
+    }
+
+    @Override
+    @Deprecated
+    public OriginalType toOriginalType() {
+      return null;
+    }
+
+    @Override
+    public <T> Optional<T> accept(LogicalTypeAnnotationVisitor<T> logicalTypeAnnotationVisitor) {
+      return logicalTypeAnnotationVisitor.visit(this);
+    }
+
+    @Override
+    LogicalTypeToken getType() {
+      return LogicalTypeToken.GEOMETRY;
+    }
+
+    @Override
+    protected String typeParametersAsString() {
+      StringBuilder sb = new StringBuilder();
+      sb.append("(").append(crs != null && !crs.isEmpty() ? crs : "").append(")");
+      return sb.toString();
+    }
+
+    public String getCrs() {
+      return crs;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (!(obj instanceof GeometryLogicalTypeAnnotation)) {
+        return false;
+      }
+      GeometryLogicalTypeAnnotation other = (GeometryLogicalTypeAnnotation) obj;
+      return Objects.equals(crs, other.crs);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(crs);
+    }
+
+    @Override
+    PrimitiveStringifier valueStringifier(PrimitiveType primitiveType) {
+      return PrimitiveStringifier.WKB_STRINGIFIER;
+    }
+  }
+
+  public static class GeographyLogicalTypeAnnotation extends LogicalTypeAnnotation {
+    private final String crs;
+    private final EdgeInterpolationAlgorithm edgeAlgorithm;
+
+    private GeographyLogicalTypeAnnotation(String crs, EdgeInterpolationAlgorithm edgeAlgorithm) {
+      this.crs = crs;
+      this.edgeAlgorithm = edgeAlgorithm;
+    }
+
+    @Override
+    @Deprecated
+    public OriginalType toOriginalType() {
+      return null;
+    }
+
+    @Override
+    public <T> Optional<T> accept(LogicalTypeAnnotationVisitor<T> logicalTypeAnnotationVisitor) {
+      return logicalTypeAnnotationVisitor.visit(this);
+    }
+
+    @Override
+    LogicalTypeToken getType() {
+      return LogicalTypeToken.GEOGRAPHY;
+    }
+
+    @Override
+    protected String typeParametersAsString() {
+      StringBuilder sb = new StringBuilder();
+
+      boolean hasCrs = crs != null && !crs.isEmpty();
+      boolean hasEdgeAlgorithm = edgeAlgorithm != null;
+
+      if (!hasCrs && !hasEdgeAlgorithm) {
+        return ""; // Return empty string when both are empty
+      }
+
+      sb.append("(");
+      if (hasCrs) {
+        sb.append(crs);
+      }
+      if (hasEdgeAlgorithm) {
+        if (hasCrs) sb.append(",");
+        sb.append(edgeAlgorithm);
+      }
+      sb.append(")");
+      return sb.toString();
```

Comment:
```suggestion
      boolean hasCrs = crs != null && !crs.isEmpty();
      boolean hasAlgo = algorithm != null;
      if (!hasCrs && !hasAlgo) {
        return "";
      }
      return String.format("(%s,%s)", hasCrs ? crs : DEFAULT_CRS, hasAlgo ? algorithm : DEFAULT_ALGO);
```

Let's be more readable. This requires us to define following variables somewhere at the top:

```
  public static final String DEFAULT_CRS = "OGC:CRS84";
  public static final EdgeInterpolationAlgorithm DEFAULT_ALGO = EdgeInterpolationAlgorithm.SPHERICAL;
```